cpp/src/slice2swift/Gen.cpp

```diff
@@ -156,29 +148,19 @@ Gen::ImportVisitor::visitClassDefStart(const ClassDefPtr& p)
 bool
 Gen::ImportVisitor::visitInterfaceDefStart(const InterfaceDefPtr& p)
 {
-    //
     // Add imports required for base interfaces
-    //
     InterfaceList bases = p->bases();
     for (const auto& base : bases)
     {
-        addImport(dynamic_pointer_cast<Contained>(base), p);
+        addImport(base, p);
     }
 
-    //
     // Add imports required for operation parameters and return type
-    //
-    const OperationList operationList = p->allOperations();
-    for (const auto& i : operationList)
+    for (const auto& operation : p->allOperations())
     {
-        const TypePtr ret = i->returnType();
-        if (ret && ret->definitionContext())
-        {
-            addImport(ret, p);
-        }
+        addImport(operation->returnType(), p);
```

Comment:
Why don't we need to check `ret->definitionContext()` anymore?

Reply:
Because:
`dynamic_pointer_cast<Contained>(p) == true`
implies that
`p->definitionContext() != null`

Both of these are about checking the same thing: "is this a user-defined type?"

Because any user defined check will have a `definitionContext` (as opposed to something like `int`, a built-in type).
And will also be a subclass of `Contained` in the compiler.